Core/include/Acts/AmbiguityResolution/GreedyAmbiguityResolution.ipp

```diff
@@ -0,0 +1,74 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/AmbiguityResolution/GreedyAmbiguityResolution.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t, typename source_link_hash_t,
```

Comment:
Since these functions are not stored only passed to the function, I guess in this case templates do make sense.